parsing/location.ml

```diff
@@ -663,6 +663,11 @@ type report_printer = {
     Format.formatter -> (Format.formatter -> unit) -> unit;
 }
 
+type json_report_printer = Misc.Log.json_fragments -> report -> unit
+
+type full_report_printer =
+  { direct: report_printer; json: json_report_printer }
```

Comment:
Here the naming is a bit odd. Maybe `report_printer` could be `terminal_report_printer`? (And maybe consider renaming `Direct` to `Terminal` or `Display`?)

Reply:
Using `terminal` for the report printer makes sense, I am not completely sure for the log itself: we are sending the data directly to the underlying formatter using `Direct` doesn't seem completely alien.